### ISSUE TITLE
Better support of WeakHashMap in dominator tree/retained set

### ISSUE THREAD

**Issue by eclipsewebmaster** [eclipse-mat/org.eclipse.mat#23]
| --- | --- |
| Bugzilla Link | [470369](https://bugs.eclipse.org/bugs/show_bug.cgi?id=470369) |
| Status | NEW |
| Importance | P3 normal |
| Reported | Jun 17, 2015 08:51 EDT |
| Modified | Feb 27, 2020 06:10 EDT |
| Version | 1.4 |
| Reporter | Vladimir Sitnikov |

## Description 
Currently MAT does not recognize the fact that "WeakHashMap" is not the only dominator of the values that are stored in the map.

In other words, it might realize that if all strong reference to the key are gone, the value would go away.

It makes sense to support that in calculation of dominator tree.\
Additional support might be needed for some well-known weak/soft collections.
